forum/profiles/views.py

```diff
@@ -37,17 +42,76 @@ def perform_create(self, serializer):
         serializer.save(user=self.request.user)
 
 
-class SaveStartupViewSet(GenericViewSet):
-    """Adds startup to user's favourites"""
+class SaveStartupViewSet(ListModelMixin, GenericViewSet):
+    """Managing user's favourite startups"""
     permission_classes = [IsAuthenticated]
-    serializer_class = Serializer
-    queryset = StartupProfile.objects.all()
+    serializer_class = StartupProfileSerializer
+
+    filter_backends = [DjangoFilterBackend, SearchFilter, OrderingFilter]
+
+    search_fields = ['company_name', 'industry', 'country', 'city']
+    filterset_fields = ['industry', 'country', 'city', 'size']
+    ordering_fields = ['company_name', 'created_at']
+
+    def get_queryset(self):
+        """Returns queryset for current user's saved startups"""
+        if self.request.user.is_authenticated:
+            investor = get_object_or_404(InvestorProfile, user=self.request.user)
+            return investor.followed_startups.all()
```

Comment:
Consider adding a `select_related` or `prefetch_related` for `followed_startups` to reduce database queries when accessing related fields.

Reply:
prefetching projects and their description too.